include/clspv/MemFence.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 The Clspv Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CLSPV_INCLUDE_CLSPV_MEM_FENCE_H_
```

Comment:
A separate header is ok, but it probably belongs in lib/ and not as part of the exported headers of the project.

Reply:
I have moved it, and added it inside a namespace for more clarification, let me know if you think that it is not necessary for an additional namespace for that.